TermsGroupSourceTests.testSupportsIncrementalBucketUpdate is flaky

**Build scan**:
https://gradle-enterprise.elastic.co/s/et6j32wppwsqk

**Repro line**:
./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.transform.transforms.pivot.TermsGroupSourceTests.testSupportsIncrementalBucketUpdate" -Dtests.seed=8ED60956387B79DC -Dtests.security.manager=true -Dtests.locale=ca-ES -Dtests.timezone=America/Monterrey -Druntime.java=11

**Reproduces locally?**: yes

**Applicable branches**: master, 7.x

**Failure excerpt**:
```
  2> java.lang.AssertionError: expected:<false> but was:<true>
        at __randomizedtesting.SeedInfo.seed([8ED60956387B79DC:EA838C26671EEB47]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.elasticsearch.xpack.core.transform.transforms.pivot.TermsGroupSourceTests.testSupportsIncrementalBucketUpdate(TermsGroupSourceTests.java:53)
```

I encountered this test failing in a PR, and found it reproduced on master, inconsistently. I ran with 20 iterations and got 8 failures.

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/pivot/TermsGroupSourceTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/pivot/TermsGroupSourceTests.java
@@ -48,10 +48,4 @@ public class TermsGroupSourceTests extends AbstractSerializingTestCase<TermsGrou
         return TermsGroupSource::new;
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/60794")
-    public void testSupportsIncrementalBucketUpdate() {
-        TermsGroupSource terms = randomTermsGroupSource();
-        assertEquals(terms.getScriptConfig() == null, terms.supportsIncrementalBucketUpdate());
-    }
-
 }

